multiview/embed/splitae.py

```diff
@@ -0,0 +1,257 @@
+        self.view1_encoder_ = self.view1Encoder_
+        self.view1_decoder_ = self.view1Decoder_
+        self.view2_decoder_ = self.view2Decoder_
+
+        if self.printGraph:
+            print("Parameter counts: \nview1Encoder: {:,}\nview1Decoder: {:,}"
+                  "\nview2Decoder: {:,}"
+                  .format(self.view1Encoder_.paramCount(),
+                          self.view1Decoder_.paramCount(),
+                          self.view2Decoder_.paramCount())
+                  )
+
+        parameters = [self.view1Encoder_.parameters(),
+                      self.view1Decoder_.parameters(),
+                      self.view2Decoder_.parameters()]
+        optim = torch.optim.Adam(itertools.chain(*parameters),
+                                 lr=self.learningRate)
+        nSamples = view1.shape[0]
+        epochTrainErrors = []
+        epochTestErrors = []
+
+        for epoch in tqdm.tqdm(range(self.trainingEpochs),
+                               disable=(not self.printInfo)):
+            batchErrors = []
+            for batchNum in range(nSamples // self.batchSize):
+                optim.zero_grad()
+                view1Batch = view1[batchNum*self.batchSize:
+                                   (batchNum+1)*self.batchSize]
+                view2Batch = view2[batchNum*self.batchSize:
+                                   (batchNum+1)*self.batchSize]
+                embedding = self.view1Encoder_(view1Batch.to(device))
+                view1Reconstruction = self.view1Decoder_(embedding)
+                view2Reconstruction = self.view2Decoder_(embedding)
+                view1Error = torch.nn.MSELoss()(view1Reconstruction,
+                                                view1Batch.to(device))
+                view2Error = torch.nn.MSELoss()(view2Reconstruction,
+                                                view2Batch.to(device))
+                totalError = view1Error + view2Error
+                totalError.backward()
+                optim.step()
+                batchErrors.append(totalError.item())
+            if self.printInfo:
+                print("Average train error during epoch {} was {}"
+                      .format(epoch, np.mean(batchErrors)))
+            epochTrainErrors.append(np.mean(batchErrors))
+            if validation_Xs is not None:
+                testError = self._testError(validation_Xs)
+                if self.printInfo:
+                    print("Average test  error during epoch {} was {}\n"
+                          .format(epoch, testError))
+                epochTestErrors.append(testError)
+
+        if self.printGraph:
+            plt.plot(epochTrainErrors, label="train error")
+            if validation_Xs is not None:
+                plt.plot(epochTestErrors, label="test error")
+            plt.title("Errors during training")
+            plt.xlabel("Epoch")
+            plt.ylabel("Error")
+            plt.legend()
+            plt.show()
+
+    def _testError(self, Xs):
```

Comment:
need documentation here

Reply:
sure, should be pretty self explanatory though